views/js/qtiCreator/editor/response/choiceSelector.js

```diff
@@ -16,39 +16,110 @@
-        resetChoices : function(){}
+        getChoices : function(){
+            return this.config.interaction.choices || {};
+        },
+        getSelectedChoices : function() {
+            return this.config.choices || [];
+        },
+        setSelectedChoices : function(choices) {
+            this.config.choices = choices;
+        }
     };
-    
-    function init(){
-        //@todo do some magic here
-        var selectedChoices = this.config.choices || [];
-        var availableChoices = this.config.interaction.getChoices();
-        console.log('selectedChoices', selectedChoices);
-        console.log('availableChoices', availableChoices);
+
+    /**
+     * Format option for select2 usage
+     *
+     * @param state
+     * @returns {string}
+     */
+    function formatOption (state) {
+        return '<span title="' + $(state.element).attr('title') + '">' + state.text + '</span>';
     }
-    
-    function destroy(){
-        //@todo clean up
+
+    /**
+     * Add a title to select2 options
+     *
+     * @param {String} content
+     * @param {Number} threshold
+     * @returns {*}
+     */
+    function createOptionTitle (content, threshold) {
+        var fullText = $('<div>', { html: content }).text().trim().replace(/\s+/g, ' ');
+        var shortText = fullText.substr(0, threshold);
+        return fullText.length - shortText.length <= 5 ? fullText : shortText + '…';
     }
-    
-    var choiceSelectorFactory = function breadcrumbsFactory(config) {
+
+
+    /**
+     * Set some additional parameters
+     */
+    var init = function init(){
+        if(!this.config.titleLength) {
+            this.config.titleLength = 30;
+        }
+    };
+
+
+    /**
+     * Select 2 needs to be removed prior to destruction of the component
+     */
+    var destroy = function destroy(){
+        this.$component.find('select').select2('destroy');
+    };
+
+
+    /**
+     * Populate select box and apply select2
+     */
+    var render = function postRender() {
+
+        var self = this;
+        var $selectBox = this.$component.find('select');
+        var selectedChoices = self.getSelectedChoices();
+
+        _.forOwn(self.getChoices(), function(valueObj, key) {
+            var option = new Option(valueObj.attr('identifier'), key, selectedChoices.indexOf(key) > -1);
+            option.title = createOptionTitle(valueObj.bdy.bdy, self.config.titleLength);
+            $selectBox.append(option);
+        });
+
+        $selectBox.select2({
+            dropdownAutoWidth: true,
+            placeholder: $selectBox.attr('placeholder'),
+            minimumResultsForSearch: -1,
+            formatResult: formatOption,
+            formatSelection: formatOption
+        }).on('change', function() {
+            self.setSelectedChoices($selectBox.select2('val'));
+            self.trigger('change', self.getSelectedChoices());
+        });
+    };
+
+    var choiceSelectorFactory = function choiceSelectorFactory(config) {
```

Comment:
please give the specific config options available to your component in the js doc of the factory method